crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1004 @@
+use std::{
+    borrow::Cow,
+    fmt::{Debug, Write},
+    result::Result,
+};
+
+use bevy_utils::{
+    tracing::{error, warn},
+    HashMap, HashSet,
+};
+
+use fixedbitset::FixedBitSet;
+use petgraph::{algo::tarjan_scc, prelude::*};
+
+use crate::{
+    self as bevy_ecs,
+    schedule_v3::*,
+    system::{BoxedSystem, Resource},
+    world::World,
+};
+
+/// TBD
+pub struct Schedule {
+    graph: ScheduleMeta,
+    executable: SystemSchedule,
+    executor: Box<dyn SystemExecutor>,
+}
+
+impl Schedule {
+    pub fn new() -> Self {
+        Self {
+            graph: ScheduleMeta::new(),
+            executable: SystemSchedule::new(),
+            executor: Box::new(MultiThreadedExecutor::new()),
+        }
+    }
+
+    pub fn add_system<P>(&mut self, system: impl IntoSystemConfig<P>) {
+        self.graph.add_system(system);
+    }
+
+    pub fn configure_set(&mut self, set: impl IntoSystemSetConfig) {
+        self.graph.configure_set(set);
+    }
+
+    pub fn set_default_set(&mut self, set: impl SystemSet) {
+        self.graph.set_default_set(set);
+    }
+
+    pub fn set_executor(&mut self, executor: ExecutorKind) {
+        self.executor = match executor {
+            ExecutorKind::Simple => Box::new(SimpleExecutor::new()),
+            ExecutorKind::SingleThreaded => Box::new(SingleThreadedExecutor::new()),
+            ExecutorKind::MultiThreaded => Box::new(MultiThreadedExecutor::new()),
+        };
+    }
+
+    pub(crate) fn initialize(&mut self, world: &mut World) -> Result<(), BuildError> {
+        if self.graph.changed {
+            self.graph.initialize(world)?;
+            self.graph.update_schedule(&mut self.executable)?;
+            self.executor.init(&self.executable);
+        }
+
+        Ok(())
+    }
+
+    pub fn run(&mut self, world: &mut World) {
+        self.initialize(world).unwrap();
+        self.executor.run(&mut self.executable, world);
+    }
+
+    /// Iterates all component change ticks and clamps any older than [`MAX_CHANGE_AGE`](crate::change_detection::MAX_CHANGE_AGE).
+    /// This prevents overflow and thus prevents false positives.
+    ///
+    /// **Note:** Does nothing if the [`World`] counter has not been incremented at least [`CHECK_TICK_THRESHOLD`](crate::change_detection::CHECK_TICK_THRESHOLD)
+    /// times since the previous pass.
+    #[doc(hidden)]
+    pub fn check_change_ticks(&mut self, change_tick: u32, last_change_tick: u32) {
+        #[cfg(feature = "trace")]
+        let _span = bevy_utils::tracing::info_span!("check schedule ticks").entered();
+        todo!();
+    }
+}
+
+/// TBD
+#[derive(Resource)]
+pub struct Schedules {
+    inner: HashMap<BoxedScheduleLabel, Option<Schedule>>,
+    default: Option<BoxedScheduleLabel>,
+}
+
+impl Schedules {
+    pub fn new() -> Self {
+        Self {
+            inner: HashMap::new(),
+            default: None,
+        }
+    }
+
+    pub fn set_default(&mut self, label: impl ScheduleLabel) {
```

Comment:
Please add docs to this. The naming doesn't tell the user exactly what this does.